### ISSUE TITLE
can't build rocRAND on nvidia platform, requires amd_comgr

### ISSUE THREAD

**Issue by jakub329homola** [ROCmSoftwarePlatform/rocRAND#196]
**Describe the bug**
on nvidia platform, building rocRAND, cmake finishes with error - could not find package amd_comgr
I don't have the comgr installed, but, should I? I am on nvidia platform.

**To Reproduce**
HIP_PLATFORM=nvidia
cuda installed in standard location (/usr/local/cuda)
installed HIP to ${HOME}/apps/HIP/installation
HIP_PATH=${HOME}/apps/HIP/installation
git clone https://github.com/ROCmSoftwarePlatform/rocRAND.git (also tried the release https://github.com/ROCmSoftwarePlatform/rocRAND/archive/refs/tags/rocm-4.2.0.tar.gz)
cd rocRAND; mkdir build; cd build
CXX=nvcc cmake ..
fails, the output is:
```
-- The CXX compiler identification is GNU 9.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/local/cuda/bin/nvcc - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Setting build type to 'Release' as none was specified.
-- Found Git: /usr/bin/git (found version "2.17.1")
-- Performing Test HAVE_gfx803
-- Performing Test HAVE_gfx803 - Failed
-- Performing Test HAVE_gfx900:xnack-
-- Performing Test HAVE_gfx900:xnack- - Failed
-- Performing Test HAVE_gfx906:xnack-
-- Performing Test HAVE_gfx906:xnack- - Failed
-- Performing Test HAVE_gfx908:xnack-
-- Performing Test HAVE_gfx908:xnack- - Failed
-- Performing Test HAVE_gfx90a:xnack-
-- Performing Test HAVE_gfx90a:xnack- - Failed
-- Performing Test HAVE_gfx90a:xnack+
-- Performing Test HAVE_gfx90a:xnack+ - Failed
-- Performing Test HAVE_gfx1030
-- Performing Test HAVE_gfx1030 - Failed
CMake Error at /home/jakub/apps/cmake-3.20.5/share/cmake-3.20/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  By not providing "Findamd_comgr.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "amd_comgr", but CMake did not find one.

  Could not find a package configuration file provided by "amd_comgr" with
  any of the following names:

    amd_comgrConfig.cmake
    amd_comgr-config.cmake

  Add the installation prefix of "amd_comgr" to CMAKE_PREFIX_PATH or set
  "amd_comgr_DIR" to a directory containing one of the above files.  If
  "amd_comgr" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  /home/jakub/apps/HIP/installation/lib/cmake/hip/hip-config.cmake:155 (find_dependency)
  cmake/VerifyCompiler.cmake:27 (find_package)
  CMakeLists.txt:110 (include)


-- Configuring incomplete, errors occurred!
See also "/home/jakub/apps/rocRAND/build/CMakeFiles/CMakeOutput.log".
See also "/home/jakub/apps/rocRAND/build/CMakeFiles/CMakeError.log".
```

**Expected behavior**
I expected rocRAND to not require amd_comgr when building for nvidia platform and to build successfully without it.

**Environment**
Ubuntu-18.04 in WSL2 with working cuda
cmake 3.20.5
hipcc --version:
```
HIP version: 4.2.21155-37cb3a34
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2021 NVIDIA Corporation
Built on Wed_Jun__2_19:15:15_PDT_2021
Cuda compilation tools, release 11.4, V11.4.48
Build cuda_11.4.r11.4/compiler.30033411_0
```

**in conclusion:**
is this really the HIP equivalent of cuRAND, or is it just the rocm equivalent working on amd platform? the README says the HIP wrapper is here, so I am confused.
I found hipBLAS, hipSPARSE, hipFFT, but not hipRAND only this rocRAND. is this the one?
how should I get it working on nvidia platform?
do I really need amd_comgr and its own requirements, like AMDDeviceLibs, even on the nvidia platform?
am I missing something fundamental here?

I am trying to build it from source, using apt-get is the last option for me.

Thanks for any help.
Jakub

PS: all the installation/build guides all around HIP should me more thorough IMO. I am really struggling a lot.

**Comment by jakub329homola** [comment]
Hello,
thanks for the answer.

> It's planned to split up to hipRAND and rocRAND

That's nice to hear!

> If you want to use nvcc for build you only need hip-base and hip-nvcc packages. In our CI we do not install amd_comgr so this is a correct expectation.

I just triend installing the hip-base and hip-nvcc packages using apt-get and now rocRAND builds successfully on the nvidia platform. I then uninstalled it all again. Then I tried to build&install HIP from source (now the new the rocm-4.3.x branch), and pointed HIP_PATH to the installation directory and added the bin directory to PATH, the previously mentioned error still occurs. 

I think I just found the issue. Looking into the error message more closely, the amd_comgr dependency comes from /home/jakub/apps/HIP/installation/lib/cmake/hip/hip-config.cmake:155. For some reason cmake preffers to read that instead of /home/jakub/apps/HIP/installation/cmake/FindHIP.cmake (I am not very fluent in cmake).

Using
```
cmake -DCMAKE_MODULE_PATH="${HIP_PATH}/cmake" ..
```
solved the issue. As mentioned, I am not very good at cmake (but I am trying to understand it better), this could have been noted in the build/installation guide.

Anyway, thanks for the help.
Jakub
